client/ayon_applications/process.py

```diff
@@ -0,0 +1,682 @@
+        the `/FI` filter option.
+
+        We should refactor this method once we find out that the fallback
+        method is not needed anymore.
+
+        Args:
+            pid_triplets (list[ProcessIdTriplet]): Processes ID to check.
+
+        Returns:
+            list[tuple[int, bool]]: List of tuples with process ID and
+                boolean indicating if the process is running.
+
+        """
+        if not pid_triplets:
+            result: list[tuple[int, bool]] = []
+
+            return result
+
+        with contextlib.suppress(ImportError):
+            return ProcessManager._check_processes_running_psutil(
+                pid_triplets)
+
+        # Fallback for systems without psutil
+        if platform.system().lower() == "windows":
+            return ProcessManager._check_processes_running_win(
+                pid_triplets)
+
+        return ProcessManager._check_processes_running_unix(
+            pid_triplets)
+
+    @staticmethod
+    def _check_processes_running_psutil(
+            pid_triplets: list[ProcessIdTriplet]) -> list[tuple[int, bool]]:
+        """Check if processes are running using psutil.
+
+        Args:
+            pid_triplets (list[ProcessIdTriplet]): List of triplets
+
+        Returns:
+            list[tuple[int, bool]]: List of tuples with process ID and
+                boolean indicating if the process is running.
+
+        """
+        result: list[tuple[int, bool]] = []
+        import psutil
+        for pid, exe, start_time in pid_triplets:
+            try:
+                is_running = ProcessManager._is_process_running_psutils(
+                    pid, exe, start_time
+                )
+            except Exception:  # noqa: BLE001
+                # if something goes wrong, fall back to pid_exists
+                try:
+                    is_running = psutil.pid_exists(pid)
+                except Exception:   # noqa: BLE001
+                    is_running = False
+            result.append((pid, is_running))
+        return result
+
+    @staticmethod
+    def get_executable_path_by_pid(pid: int) -> Optional[Path]:
+        """Get the executable path of a process by its PID using psutil.
+
+        Args:
+            pid (int): Process ID.
+
+        Returns:
+            Optional[Path]: The executable path of the process, or None if it
+                cannot be determined.
+
+        """
+        try:
+            import psutil
+        except ImportError:
+            return None
+
+        exe_path = None
+        if pid:
+            try:
+                exe_path_str = psutil.Process(pid).exe()
+                if exe_path_str:
+                    exe_path = Path(exe_path_str)
+            except (
+                    psutil.NoSuchProcess,
+                    psutil.ZombieProcess,
+                    psutil.AccessDenied):
+                exe_path = None
+        return exe_path
+
+    @staticmethod
+    def _check_processes_running_win(
```

Comment:
You can look at `_windows_pid_is_running` or rather `_windows_get_pid_args` in `ayon_core/tools/tray/lib.py`. Is using windows api to get the args to avoid running subprocess and parsing of the output.